x-pack/plugins/secops/server/graphql/events/resolvers.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FieldNode, SelectionNode, SelectionSetNode } from 'graphql';
+import { isEmpty } from 'lodash/fp';
+
+import { SourceResolvers } from '../../../common/graphql/types';
+import { Events } from '../../lib/events';
+import { EventsRequestOptions } from '../../lib/events/types';
+import { AppResolvedResult, AppResolverOf } from '../../lib/framework';
+import { Context } from '../../lib/types';
+import { parseFilterQuery } from '../../utils/serialized_query';
+import { QuerySourceResolver } from '../sources/resolvers';
+
+type QueryEventsResolver = AppResolverOf<
+  SourceResolvers.GetEventsResolver,
+  AppResolvedResult<QuerySourceResolver>,
+  Context
+>;
+
+interface EventsResolversDeps {
+  events: Events;
+}
+
+export const createEventsResolvers = (
+  libs: EventsResolversDeps
+): {
+  Source: {
+    getEvents: QueryEventsResolver;
+  };
+} => ({
+  Source: {
+    async getEvents(source, args, { req }, info) {
+      const fields = getFields(info.fieldNodes[0]);
```

Comment:
Does indexing into `fieldNodes` require a `length` check?

Optional: this is totally a matter of style, but sometimes I like to use the `lodash` [first](https://lodash.com/docs/2.4.2#first) function

Reply:
I should be protected by graphql itself there